src/wallet/wallet.h

```diff
@@ -30,7 +30,8 @@
 
 #include <boost/shared_ptr.hpp>
 
-extern CWallet* pwalletMain;
+typedef CWallet* CWallet_ptr;
```

Comment:
Typedef-ing a pointer type doesn't seem that useful to me either. I suppose the rationale is "what if we want to change it to e.g. std::shared_ptr later". But a) I think the code is more readable if the kind of pointer is clear from the function/struct/class signatures using it b) You might want to have different kinds of pointers to the same kind of objects in different places.